lightningd/lightningd.c

```diff
@@ -574,15 +574,49 @@ void notify_new_block(struct lightningd *ld, u32 block_height)
 	channel_notify_new_block(ld, block_height);
 }
 
+static void on_sigint(int _ UNUSED)
+{
+        static const char *msg = "lightningd: SIGINT caught, exiting.\n";
+        write(STDERR_FILENO, msg, strlen(msg));
+        _exit(1);
+}
+
+static void on_sigterm(int _ UNUSED)
+{
+        static const char *msg = "lightningd: SIGTERM caught, exiting.\n";
+        write(STDERR_FILENO, msg, strlen(msg));
+        _exit(1);
+}
+
+/*~ Signal handlers for SIGINT and SIGTERM to print message to
+ * STDERR and exit. We only do this if we are PID 1. */
+static void setup_sig_handlers(struct sigaction *sigint,
+			       struct sigaction *sigterm)
```

Comment:
Why not just declare `sigint` and `sigterm` structures as local here? Caller does not use them anyway.